src/expunge/expunge.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+
+import collections
+import enum
+
+class Level(object):
+    """ Crime Level.
+
+    Describes crime level. e.g. Felony Class A.
+
+    Attributes:
+        type_: A string describing the type of crime.
+        class_: A string of length 1 specifying the class.
+    """
+    def __init__(self, type_, class_):
+        self.type_ = type_
+        self.class_ = class_
+
+DispositionType = enum.Enum('Disposition',
+                            ' '.join([
+                                'CONVICTION',
+                                'NO_CONVICTION',
+                                'DISMISSED',
+                                'ACQUITTED',
+                                'NO_COMPLAINT'
+                                ]))
+
+class Disposition(object):
+    """ Disposition for a charge.
+
+    Attributes:
+        type_: An enum of type DispositionType.
+        date: A datetime.date specifying the date of the disposition.
+    """
+    def __init__(self, type_, date):
+        self.type_ = type_
+        self.date = date
+
+class Statute(object):
+    """ Statute corresponding to a law
+
+    Statutes are represented by two numbers: chapter and sub-chapter delimited
+    by a period. e.g. 653.412
+
+    Attributes:
+        chapter: An integer that specifies statute chapter.
+        subchapter: An integer that specifies statute sub-chapter.
+    """
+    def __init__(self, chapter, subchapter):
+        self.chapter = chapter
+        self.subchapter = subchapter
+
+    def __str__(self):
+        # TODO do these need to have leading zeros?
+        return '.'.join([self.chapter, self.subchapter])
+
+class Charge(object):
+    """ Charge filed on a Client.
+
+    Attributes:
+        name: A string describing the charge.
+        statute: A Statute object that applies for the charge.
+        date: A datetime.date object specifying the date of the charge.
+        disposition: A Disposition object for the charge.
+        fine: A float that tells how much money is owed to the court.
+    """
+    def __init__(
+            self,
+            name,
+            statutes,
+            level,
+            date,
+            disposition,
+            fine=0):
```

Comment:
I think the fine should be moved to the `Case` class and call it `balance_due` ? The disposition page has an easy to parse balance due and that is what the client is most interested in. It will be difficult to extract the fine per charge as its not clearly stated. If there are payments to be made for the charges it's grouped together for each case at the bottom of the disposition page in the financial section.